libcontainer/rootfs_linux.go

```diff
@@ -98,6 +98,15 @@ func prepareRootfs(pipe io.ReadWriter, iConfig *initConfig) (err error) {
 		return newSystemErrorWithCausef(err, "changing dir to %q", config.Rootfs)
 	}
 
+	s := iConfig.SpecState
+	if s != nil {
+		s.Pid = unix.Getpid()
```

Comment:
This will always be equal to `1` (unless pid namespaces are disabled, obviously). Shouldn't this be the PID in the parent namespace -- which is already filled in? Then again, maybe the hook wants to know the PID of the container process in the namespace it's being run in? I'm not sure...

Reply:
My mental model is:
- Runtime namespace hooks get runtime/absolute information
- Container Namespace hooks get container/relative information

I can specify that in the hooks documentation to help with that

Reply:
Yeah that does make sense. It wouldn't hurt to mention it in the hook documentation.

Reply:
Actually reading the spec, it always  "is the ID of the container process, as seen by the host."

Reply:
Thinking about it a little more -- I think that's a bug in the spec. Yes, there isn't an `exec` primitive in the specification, but you can manually share the PID namespace which would mean that the hook run inside the container won't be able to make use of the PID in the state JSON.

Reply:
So keep `unix.Getpid()` for now and address it in the spec?

Reply:
For now, yes. But please open a runtime-spec issue so we don't lose track of it.

Reply:
https://github.com/opencontainers/runtime-spec/pull/1045